Transaction/index.ts

```diff
@@ -93,6 +142,26 @@ export namespace Transaction {
 			return (({ state, ...event }) => event)(transaction)
 		}
 	}
+	export function amountFromState(state: Rule.State.Evaluated): Transaction["amount"] {
+		const sign = ["outbound", "authorization", "capture"].some(direction => direction == state.transaction.kind)
+			? -1
+			: 1
+		return {
+			original: sign * state.transaction.original.amount,
+			reserved: -1 * (state.transaction.original.reserve ?? 0),
+			charge: -1 * (state.transaction.original.charge?.total ?? 0),
```

Comment:
I don't think we should swap signs since:
- processing: original + reserved = total
- finalized: original + charge = total